File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    void dfs(TreeNode* root,int m,int &count)
    {
    if(root==nullptr)
    {
        return;
    }
    if(root->val >= m)
    {
        count++;
        m=root->val;
    }

    dfs(root->right,m,count);
    dfs(root->left,m,count);
    }

    int goodNodes(TreeNode* root) {
        
        if(root==nullptr)
        {
            return 0;
        }
        int count = 0;
        dfs(root,root->val,count);
        return count;
        
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
	ans := make([][]int, 0)
	var dfs func(*TreeNode,int)
	dfs = func(r *TreeNode, i int){
		if r == nil{
			return
		}
	if i >= len(ans){
		ans = append(ans, []int{})
	}

	ans[i] = append(ans[i], r.Val)
	dfs(r.Left, i+1)
	dfs(r.Right, i+1)
	}
	dfs(root,0)
	return ans
}
